### ISSUE TITLE
Video playing in 720p maximum but downloading in 1080p

### ISSUE THREAD

**Issue by AndreiBiziuk** [iamtalhaasghar/yewtube#36]
## Issue / Suggestion

Any video that I try playing in 1280x720 resolution maximum despite `max_res` set to `2160p`, `video_format auto`. Downloading the same video results in 1080p file.

## Your Environment

>-> % 
yt --version
yewtube version    : 2.7.0
yt_dlp version     : 2022.02.04
Python version     : 3.10.2 (main, Jan 15 2022, 19:56:27) [GCC 11.1.0]
Processor          : 
Machine type       : x86_64
Architecture       : 64bit, ELF
Platform           : Linux-5.16.9-arch1-1-x86_64-with-glibc2.35
sys.stdout.enc     : utf-8
default enc        : utf-8
Config dir         : /home/andrei/.config/mps-youtube
env:TERM           : xterm-256color
env:SHELL          : /bin/zsh
env:LANG           : ru_RU.UTF-8
env:LANGUAGE       : ru:en_US

**Comment by iamtalhaasghar** [comment]
This is because videos having resolution greater than **720p** have separate audio and video streams. As per my knowledge we can not mux / combine them together and play at the same time in any player (including vlc).
So, when you tell `yewtube` to `show_video` it picks the best video stream having both audio codec / stream as well as video codec / stream. 
https://github.com/iamtalhaasghar/yewtube/blob/28eadb75c725ce8c2aa4d790f61f12c78b8e96a9/mps_youtube/streams.py#L71

On the other hand when you download a video, `yt-dlp` downloads both the audio and video streams separately and then use `ffmpeg` to combine them into a single file.

**tldr;** theoretically we can play 2160p videos in yewtube but with no audio / sound.

**Comment by sidoverflow** [comment]
@iamtalhaasghar I think you are right about the separate audio and video streams. however mpv is able to play youtube videos at a higher resolution (1080p and above) by setting the `--ytdl-format` flag. so i thought yewtube should be able to do the same, i tried setting this flag with `playerargs` config on yewtube for mpv but didn't seem to work. can you tell me why this wouldn't work?

**Comment by iamtalhaasghar** [comment]
> mpv is able to play youtube videos at a higher resolution (1080p and above)

Yes, you can but the video will be dumb (no sound).

**Comment by sidoverflow** [comment]
on mpv there is sound. are you able to check streaming youtube standalone on mpv?
